src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -188,50 +90,12 @@ private static function translate_theme_json_chunk( array $array_to_translate, $
 	 * @return array Returns the modified $theme_json_structure.
 	 */
 	private static function translate( $theme_json, $domain = 'default' ) {
```

Comment:
This PR replaces the existing logic to translate `theme.json` so it would be great to confirm that everything works as before. All existing unit tests pass for processing `theme.json` are still green. New tests were added to cover new cases in the extracted `translate_settings_using_i18n_schema` method that no longer use the intermediate step to run `get_fields_to_translate` which collects an alist of fields to translate.

@jorgefilipecosta and @nosolosw, I would appreciate your feedback.

Reply:
I tested theme.json translation and things are working well 👍 Nice work! We should probably also update plugin code right?

Reply:
We could update the plugin, although it works exactly the same so I consider it a low priority. I'll have to explore what is necessary to bring `variations` handling to `block.json` for older versions of WordPress supported by the Gutenberg plugin. If we need to polyfill `translate_settings_using_i18n_schema` then we can refactor the logic for `theme.json` as well.